refactor(language_server): use Message span for Diagnostic's Range

Sysix opened the pull request:
~~Now the language server can use the primary span for the diagnostic. This was not possible before.
The idea came from @camc314 in a previous PR, when I refactored away the `MessageWithPosition` struct~~

Use the message span, it is already the primary span

camc314 commented:
### Merge activity

* **Oct 15, 9:04 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Oct 15, 9:04 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).

Sysix committed:
refactor(language_server): use Message span for Diagnostic's Range (#14600)

~~Now the language server can use the primary span for the diagnostic. This was not possible before.
The idea came from @camc314 in a previous PR, when I refactored away the `MessageWithPosition` struct~~

Use the message span, it is already the primary span
